### DISCUSSION TITLE
Couldn't find attribute or association

### DISCUSSION THREAD

**Discussion by kashiftariq1997** [excid3/madmin#91] [General]
I have integrated this gem but i am facing some issue Kindly help me or give me any idea what the wrong is this.
I am getting this error:
  
ArgumentError (Couldn't find attribute or association 'id' with type '' on Coupon model

  To fix this, either:

  1. Remove 'attribute id' from app/madmin/resources/coupon_resource.rb
  2. Or add the missing attribute or association to the Coupon model
):
But on the same time i have coupon model having id attribute exists.
Further more i have set the root path in routes to 'users#index'
@excid3 
Thanks in advance.

**Comment by excid3** [comment]
Does your Coupon model not have an ID column?

**Reply by kashiftariq1997** [reply]
@excid3 Yes my coupon model has id column already but it shows me this error. On the same this is basically the primary key of coupon model.
That's the attribute of my coupon model.
 => #<Coupon id: nil, code: nil, description: nil, discount_type: "percent_discount", amount: nil, active_for_x_payments: nil, individual_use: false, product_id: nil, free_shipping: nil, times_redeemed: 0, user_id: nil, redeem_by: nil, usage_limit: nil, expiration_date: nil, status: "draft", created_at: nil, updated_at: nil, times_applied_to_cart: 0, ambassador_id: nil>

**Reply by excid3** [reply]
Strange! You'll probably have to send me an example repo so I can see it and figure out what's going on.

**Reply by kashiftariq1997** [reply]
@excid3 https://github.com/DevDuo/cbdops.git
